### DISCUSSION TITLE
In-memory local node still having issues with hardhat deployment

### DISCUSSION THREAD

**Discussion by Gilgamesh-818** [zkSync-Community-Hub/zksync-developers#68] [General] [answered]
### Environment

Testnet

### Provide a brief description of the functionality you're trying to implement.

I am unable to deploy a contract to my local in-memory node via hardhat using `yarn hardhat deploy-zksync` command. I am able to deploy via foundry-zksync but the console logging is bugged, so I can't read the logs. The following code is my hardhat.config.ts file, I've used this config to deploy contracts on the dockerized local tests and on mainnet with no issues. I changed the ethNetwork field to "" as instructed previously. 
```
import "@matterlabs/hardhat-zksync-deploy";
import "@matterlabs/hardhat-zksync-solc";

// dynamically changes endpoints for local tests
const zkSyncTestnet =
{
    url: "http://localhost:8011",
    ethNetwork:  "",
    zksync: true,
}


module.exports = {
    zksolc: {
        version: "latest", // Uses latest available in https://github.com/matter-labs/zksolc-bin/

        settings: {},
    },
    // defaults to zkSync network
    defaultNetwork: "zkSyncTestnet",

    networks: {
        hardhat: {
            zksync: true,
        },
        // load test network details
        zkSyncTestnet,
    },
    solidity: {
        version: "0.8.13",
    },
};
```

deploy.ts 
```
import { Wallet, utils } from "zksync-web3";
import * as ethers from "ethers";
import { HardhatRuntimeEnvironment } from "hardhat/types";
import { Deployer } from "@matterlabs/hardhat-zksync-deploy";

// An example of a deploy script that will deploy and call a simple contract.
export default async function (hre: HardhatRuntimeEnvironment) {
    console.log(`Running deploy script for the Executor contract`);

    // Initialize the wallet.
    const wallet = new Wallet("ac1e735be8536c6534bb4f17f06f6afc73b2b5ba84ac2cfb12f7461b20c0bbe3");

    // Create deployer object and load the artifact of the contract you want to deploy.
    const deployer = new Deployer(hre, wallet);
    const artifact = await deployer.loadArtifact("Executor");

    // Deploy this contract. The returned object will be of a `Contract` type, similarly to ones in `ethers`.
    // `greeting` is an argument for contract constructor.
    //const parsedFee = ethers.utils.formatEther(deploymentFee.toString());
    //console.log(`The deployment is estimated to cost ${parsedFee} ETH`);

    const ExecutorContract = await deployer.deploy(artifact);

    //obtain the Constructor Arguments
    console.log("constructor args:" + ExecutorContract.interface.encodeDeploy([]));
    //Show the contract info.
    const contractAddress = ExecutorContract.address;
    console.log(`${artifact.contractName} was deployed to ${contractAddress}`);

}


```

### What is the specific issue or error you're encountering?

To me, it seems like the deployer.ts script is trying to connect to an eth rpc and failing. Im not really sure what the problem is though because again the foundry implementation works without a problem. I have scoured the docs and can't find anything so any help on this would be greatly appreciated.

### Can you share the error messages or logs you're receiving, if any?

This is the error message that I am seeing in my terminal when I run `yarn hardhat deploy-zksync`.

```
Error: could not detect network (event="noNetwork", code=NETWORK_ERROR, version=providers/5.7.2)
```

### Have you made any recent changes to the contract before encountering this issue?

No

### Are there any external libraries or contracts that your contract interacts with?

N/A

### Can you provide the relevant portions of your contract code where the issue is occurring?

N/A

### Have you tried to isolate the problem, and if so, what were the results?

N/A

### What steps have you already taken to try to resolve the issue?

N/A

**Comment by MexicanAce** [comment]
Strange error. Your hardhat configuration and deploy script look fine to me. One thing I could think of is to ensure that your local `era_test_node` is running prior to trying the deployment (but you said that you're able to use `zksync-foundry` and connect it to your `era_test_node` so perhaps it already is)

Don't know if this will have an impact or not, but the version of `solidity` that's on my local hardhat config is a higher version:
```
solidity: {
    version: "0.8.17",
  },
```

**Comment by MexicanAce** [comment]
Hmmm, when using the basic [hardhat template](https://github.com/matter-labs/zksync-hardhat-template) for zkSync , things are working with no additional changes than updating the port number of the endpoint.

Is there any way you could upload your repo (or a repo) that I could use to reproduce the issue?

**Reply by idea404** [reply]
@Gilgamesh-818 could you please respond, will otherwise close this as inactive

**Comment by uF4No** [comment, marked as answer]
Just to add a little bit more context for future reference, the `Deployer` class from `@matterlabs/hardhat-zksync-deploy` uses the RPC endpoint defined in your `hardhat.config.ts` file. Make sure the port is correct and that the node started correctly.

**Reply by MexicanAce** [reply]
As we're unable to replicate this issue with the provided code, we're closing this issue for now.

If this issue persists, perhaps change the `localhost` endpoint in the `hardhat.config.ts` to `127.0.0.1`, as that can be a problem sometimes with things like GitHub Virtual Environments.

**Comment by bradfordbradv** [comment]
It seems like the issue might be related to the configuration of the hardhat-zksync-deploy plugin. Double-check the configuration and ensure that the network settings are correctly specified, especially for the zkSyncTestnet network. Also, verify that the local in-memory node is running and accessible at the specified URL http://localhost:8011. If the issue persists, consider reaching out to the plugin maintainers or checking their documentation for troubleshooting steps.

**Comment by bartlettdv** [comment]
Environment: Testnet
Issue: Unable to deploy a contract to local in-memory node via hardhat using `yarn hardhat deploy-zksync` command.
Error: "Error: could not detect network (event='noNetwork', code=NETWORK_ERROR, version=providers/5.7.2)"
Potential Solution: Check network configuration in hardhat.config.ts, ensure correct RPC endpoint for local node, verify network connectivity.

**Comment by tathimustingjason** [comment]
It seems like the issue might be related to the network configuration in your Hardhat setup. Double-check the network configuration and ensure that the connection to the local in-memory node is properly configured.

**Comment by spatounmas** [comment]
It seems like the issue might be related to the configuration or network setup. Ensure that the local zkSync node is running and properly configured to accept deployments. Check the network configuration in your hardhat config file and verify that it matches the settings of your local zkSync node. Additionally, review any logs or error messages from the zkSync node to pinpoint the exact cause of the deployment failure.
